File: SRM684Div1/250/solution.py
class CliqueParty():
    def maxsize(self, a, k):
        a = list(a)
        a.sort()
        n = len(a)
        ans = 1
        for i in range(n):
            for j in range(i + 1, n):
                last = a[i]
                rlen = a[j] - a[i]
                count = 2

                for m in range(i + 1, j):
                    if (a[m] - last) * k >= rlen and (a[j] - a[m]) * k >= rlen:
                        last = a[m]
                        count += 1
                ans = max(ans, count)
        return ans

File: SRM684Div1/250/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class CliqueParty {
public:
    int maxsize(vector<int> a, int k) {
        sort(a.begin(), a.end());
        
        int res = 1;
        int n = a.size();
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                // Only when we fixed the last number and the first number
                // we can know the maximum number in the set is a[j] - a[i].
                // then we can only check the neighbors didn't have a small gap to satisfy the
                // condition.
                int last = a[i];
                int dist = a[j] - a[i];
                int cnt = 2;
                for (int m = i + 1; m < j; m++) {
                    // This is a greedy selection algorithm.
                    if ((a[m] - last) * k >= dist && (a[j] - a[m]) * k >= dist) {
                        cnt++;
                        last = a[m];
                    }
                }
                res = max(res, cnt);
            }
        }
        return res;
    }
};
